AlgebraInLean/Chapter01/Sheet05.lean

```diff
@@ -0,0 +1,48 @@
+import AlgebraInLean.Chapter01.Sheet04
+
+namespace AlgebraInLean
+
+/-
+To represent Cₙ, the group of rotational symmetries of a regular n-gon, we will use a type already
+in Mathlib called `Fin`. `Fin n` is the type of all natural numbers less than n. Elements are
+represented as a pair of a natural number and a proof that it is less than n. A value `k : Fin n`
+represents a rotation by 2kπ/n radians. Like the integers, this group is also abelian.
+-/
+instance (n : ℕ) [NeZero n] : AbelianGroup (Fin n) where
```

Comment:
Can you mention why we use the NeZero type-class instead of having a prop (h : $n \neq 0$)?

Reply:
That would be a question for @justin-t-morrill and @AdamJasonKern. I just removed the unused naming and made it a `[]` argument.